cpp/source/basic.rst

```diff
@@ -41,9 +41,9 @@ tedious:
   :caption: Checking the status of every function manually
   :dedent: 2
 
-The macro :c:macro:`ARROW_RETURN_NOT_OK` will take care of some of this
-boilerplate for you.  It will run the contained expression and check the resulting
-``Status`` or ``Result`` object.  If it failed then it will return the failure.
+The macros :c:macro:`ARROW_RETURN_NOT_OK` and :c:macro:`ARROW_ASSIGN_OR_RAISE` will take care of some of these
+boilerplates for you.  They will run the contained expression and check the resulting
+``Status`` or ``Result`` object.  If they failed then they will return the failure.
```

Comment:
For clarity, I think Status and Result should be introduced separately

Reply:
Sure. Should I split the section "Working with Status and Result" into two or should I keep it but move Result related stuff to a separate paragraph?

Reply:
I think one section is OK, just have the text separate so it's clear what applies to what

Reply:
(And separate code blocks)